File: projects/go-sdk/handlers/v4/updateEvent.go
package handlersv4

import (
	"encoding/json"
	"go-sdk/fingerprintv4"
	"net/http"
	"strconv"

	fingerprint "github.com/fingerprintjs/go-sdk/v8"
)

type updateEventQueryParams struct {
	ApiKey  string `json:"api_key"`
	Region  string `json:"region"`
	EventID string `json:"event_id"`
}

func UpdateEvent(w http.ResponseWriter, r *http.Request) {
	var suspectValue *bool

	query := r.URL.Query()

	suspect, suspectErr := strconv.ParseBool(query.Get("suspect"))
	if suspectErr == nil {
		suspectValue = &suspect
	}

	var tag map[string]interface{}
	json.Unmarshal([]byte(query.Get("tags")), &tag)

	linkedId := query.Get("linked_id")

	eventUpdate := fingerprint.EventUpdate{
		LinkedID: &linkedId,
		Suspect:  suspectValue,
		Tags:     tag,
	}

	queryParams := updateEventQueryParams{
		ApiKey:  query.Get("api_key"),
		Region:  query.Get("region"),
		EventID: query.Get("event_id"),
	}

	client := fingerprintv4.Init(queryParams.ApiKey, queryParams.Region)

	httpRes, err := client.UpdateEvent(r.Context(), queryParams.EventID, eventUpdate)
	response := fingerprintv4.ProcessResponse(nil, httpRes, err)

	w.Header().Set("Content-Type", "application/json")
	w.WriteHeader(200)
	json.NewEncoder(w).Encode(response)
}

File: projects/python-sdk/handlers/v4/update_event.py
from fingerprint_server_sdk import EventUpdate
from flask import request, json

from handlers.v4.fingerprint_client import create_client, create_response


def update_event():
    event_id = request.args.get('event_id', '')
    tags = request.args.get('tags')
    if tags is not None:
        tags = json.loads(tags)

    suspect = request.args.get('suspect')
    if suspect is not None:
        suspect = suspect.lower() in ['true', '1']

    event_update = EventUpdate(
        linked_id=request.args.get('linked_id'),
        suspect=suspect,
        tags=tags
    )

    api_instance = create_client()
    return create_response(lambda: api_instance.update_event_with_http_info(event_id, event_update))
